Sources/Widgets/SVG/SVGCustomLandmarkRepresentation/index.js

```diff
@@ -0,0 +1,85 @@
+import macro from 'vtk.js/Sources/macro';
+import vtkSVGRepresentation from 'vtk.js/Sources/Widgets/SVG/SVGRepresentation';
+
+const { createSvgElement } = vtkSVGRepresentation;
+
+// ----------------------------------------------------------------------------
+// vtkSVGCustomLandmarkRepresentation
+// ----------------------------------------------------------------------------
+
+function vtkSVGCustomLandmarkRepresentation(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkSVGCustomLandmarkRepresentation');
+
+  publicAPI.render = () => {
+    const list = publicAPI.getRepresentationStates();
+
+    const coords = [];
+    for (let i = 0; i < list.length; i++) {
+      coords.push(list[i].getOrigin());
+    }
+
+    return publicAPI.worldPointsToPixelSpace(coords).then((pixelSpace) => {
+      const points2d = pixelSpace.coords;
+      const winHeight = pixelSpace.windowSize[1];
+
+      const root = createSvgElement('g');
+
+      for (let i = 0; i < points2d.length; i++) {
+        const xy = points2d[i];
+        const x = xy[0];
+        const y = winHeight - xy[1];
+
+        const text = createSvgElement('text');
+        Object.keys(model.textProps || {}).forEach((prop) =>
+          text.setAttribute(prop, model.textProps[prop])
+        );
+        text.setAttribute('x', x);
+        text.setAttribute('y', y);
+        text.textContent = publicAPI
+          .getInputData()
+          .getAllNestedStates()[3]
```

Comment:
Why do you take the third one ? How do you know ? Is this class specific to the line widget or is it a class that can be reused ?

Reply:
here the 3rd one refers to the text state from the SVGCustomLandMark class. This part is hard-code for line widget since the 3 1st elements of this tab refer to handle 1, handle2 and polyline. I know this because this is the lst element to be generated in the behavior of the widget.

Reply:
Can you add a comment to explain it please ?